### PR TITLE
WIP: Updated Scale Testing Instructions, Results, And POC Fixes

### PR DESCRIPTION
**Pull request by dgoodwin**

### CODE REVIEW

**Comment by dgoodwin** [comment]
Includes #1049

File: hack/scaletest/README.md [hunk 1]
```diff
@@ -4,29 +4,114 @@ Simulate 1000 clusters in Hive and observe performance. Because we do not have t
...
-oc process -f hack/scaletest/syncset-template.yaml RESOURCE_INDEX=0 | oc apply -f -
+
+Get the cluster's admin kubeconfig which will be used to adopt this one cluster into thousands of duplicates, all backed by the same real spoke cluster:
+
+```bash
+./hack/get-kubeconfig.sh scaletest-spoke > scaletest-spoke.kubeconfig
 ```
 
-Setup a transient prometheus pod in the hive namespace (see developer docs) and used the following queries to monitor Hive behavior:
+Duplicate the real cluster by adopting copies to simulate cluster load in Hive. This can be run in multiple terminals to parallelize and generate more load, just use different start/end blocks in each.
```
Comment (csrwng) [inline hack/scaletest/README.md:27]: you can do the same by running the test_setup.sh script in the background, like:
```
$ time hack/scaletest/test_setup.sh cluster.kubeconfig 1 100 &
$ time hack/scaletest/test_setup.sh cluster.kubeconfig 101 200 &
...
```

File: hack/scaletest/README.md [hunk 2]
```diff
@@ -4,29 +4,114 @@ Simulate 1000 clusters in Hive and observe performance. Because we do not have t
...
+  
+## Recommendations
+
+  * HIGH: Patch resource/apply.go to use server side apply if the cluster is 4.5+. Test SyncSets carefully to make sure resources are getting applied the way we would expect.
+  * HIGH: Patch resource/apply.go to use in-memory cache instead of a temp file, see how this performs vs server side apply. (Cesar a good point of contact for how to proceed with this)
+  * HIGH: Switch Hive deployments to cpu optimized instance types with a lot of vcpus. I saw usage up to about 14 cores with 100 goroutines for ssi controller. We might be able to push this further.
+  * HIGH: Make syncsetinstance controller goroutines configurable in HiveConfig. (maybe for all controllers?) This is applied in syncsetinstance_controller.go AddToManager.
+  * MED: Test even more than 100 goroutines for syncsetinstance controller. Add uuids to contextual controller loggers to help debug logs when we need to.
+  * MED: Make hive-controllers cpu/memory requests configurable in HiveConfig.
+  * LOW: Consider splitting syncset instance controller into a separate pod to isolate logging and load and prevent resource contention between controllers.
```
Comment (csrwng) [inline hack/scaletest/README.md:111]: this could actually be tested today with the includes/excludes flag

**Comment by dgoodwin** [comment]
Most of this was focused on just getting the syncsetinstance controller into a steady state where it could process a full sync of everything in a reasonable time. Once we get that solved I'd like to be exploring the length of time it's taking new clusters to get all their selectorsyncsets, and how other controllers in general are behaving.

**Comment by dgoodwin** [comment]
Server side apply patch broken out into: https://github.com/openshift/hive/pull/1052

**Pull request closed without merging**